### ISSUE TITLE
Error "TS2335" false positive

### ISSUE THREAD

**Issue by AFatNiBBa** [microsoft/TypeScript#60669]
### 🔎 Search Terms

- ts2335
- static
- hasInstance
- 'super' can only be referenced in a derived class

### 🕗 Version & Regression Information

- This has NOT changed
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about "TS2335"
- I tested this on "Nightly" and "3.3.3" on the playground

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=99&ts=5.8.0-dev.20241203#code/FAYwNghgzlAEByB7ATgWwmAwpGsDewsRsALogMonICWAdgOYAUAlPocR8gKYkCuytWAEZYAalhReABy7IAdGUo0GLANzsiAX1gB6HRwNFAiYSwA5JJnJTsEBEGJaYAJ6wARl1jcAZrK60QXAAmsHSwELCBstQAbkE2OFBywJrAoAmw5DIg1BjY0HAEAJBQJBAk1CCwANrkTqiuiGByABbQAJK0JXYBALqMiK4AVgBcsLy0ANa0iADutKxFhdx8ArADg7AAvNukzdSJUsiIZCROMrAAPhcS0rI1dQ1NrVAdXf5cfevM6oXaeoYAwFA4gmcy3Kw2OxrRwudyeLg+bjvYKhcKRGixYLgfJJFKgBxQRpcORgRBMAgcdajPDaOhvAKILwZLI5LAJAA0wK5AP+XgwUC4GhCnVGtC4M2ZXGyuQSLGF9K4jMl0rZ+U5XP+VF4go4h2OiFGmSlrLyMDkepOZw8dNK7yVRpVpqg6o4muQ2qFFsQAH1pmhRkg0DKcV7Tucbd1FUzA+hVTAXbz+TriCVkKLxQgULGnXLxAAiPPcouA-55oRVdZSkiwADyQyrPTzyW+QA

### 💻 Code

```ts

class NormalClass {
  toString() {
    return 1 + super.toString();
  } //         ↑ 'super' can only be referenced in a derived class.
}

class SpecialClass {
  static [Symbol.hasInstance](obj: unknown) {
    return obj === this.prototype || super[Symbol.hasInstance](obj);
  } //                               ↑ 'super' can only be referenced in a derived class.
}

console.log({
  obj: {} instanceof SpecialClass,                            // false
  ins: new SpecialClass() instanceof SpecialClass,            // true
  proto: SpecialClass.prototype instanceof SpecialClass,      // true
  proto_norm: NormalClass.prototype instanceof NormalClass,   // false
  str: new NormalClass() + ""                                 // "1[object Object]"
});
```

### 🙁 Actual behavior

The error "TS2335" is shown for the two marked lines of code

### 🙂 Expected behavior

The code works as a charm when executed, so I think the error shouldn't come up in those cases, since is probably a false positive

**Comment by MartinJohns** [comment]
If this is intentional you can write `extends Object`. Otherwise the inheritance chain with `Object` is hidden.

**Comment by AFatNiBBa** [comment]
I don't think it's fully a duplicate because it's not only a problem with instance members (`Object.prototype.toString()`), but also static ones (`Function.prototype[Symbol.hasInstance]()`).

Moreover, `extends Object` has real runtime effects, for instance it adds the static members of `Object` to the class.
> The prototype of base classes is `Function.prototype`, if you explicitly extended `Object`, it would be the prototype instead
```js
console.log((class { }).__proto__ === Function.prototype)     // true
console.log((class extends Object { }).__proto__ === Object)  // true
```

**Comment by RyanCavanaugh** [comment]
It's still intentional to not allow any `super` access without an explicit extends clause. You can write `class Foo extends Function` if that's important in your code.

**Comment by AFatNiBBa** [comment]
As I said, it's completely different:
```js
console.log((class { }).__proto__ === Function.prototype)     // true
console.log((class { }).__proto__ === Function)               // false
```
Notice the `prototype`, the thing you said would make the INSTANCE of the class extend function.
That error makes never sense in these situations, since base members (`Object.prototype` for instance members and `Function.prototype` for static ones)
